src/esp/physics/bullet/BulletRigidObject.h

```diff
@@ -56,17 +54,20 @@ class BulletRigidObject : public BulletBase,
   /**
    * @brief Finalize this object with any necessary post-creation processes.
    */
-  virtual void finalizeObject() override;
+  void finalizeObject_LibSpecifc() override;
 
   /**
    * @brief Instantiate a bullet primtive appropriate for the passed
    * AbstractPrimitiveAttributes object
-   * @param primAttributes the AbstractPrimitiveAttributes describing the
-   * desired object
+   * @param primTypeVal int value corresponding to assets::PrimObjTypes enum
+   * describing primitive collision shape.
+   * @param halfLength half length of object, for primitives using this value
    * @return a unique pointer to the bullet primitive object
    */
   std::unique_ptr<btCollisionShape> buildPrimitiveCollisionObject(
-      const assets::AbstractPrimitiveAttributes& primAttributes);
+      int primTypeVal,
+      double halfLength);
```

Comment:
Why are we pulling back from directly consuming `AbstractPrimitiveAttributes` here? Seems like this change introduces unnecessary complexity (e.g. need to pass `halfLength` even for types which don't consume this).

Reply:
Actually gets rid of a copy of the attributes.  Isn't needed until after we actually make a copy in the rigid object initialize